Doc/library/asyncio-sync.rst

```diff
@@ -340,6 +341,132 @@ BoundedSemaphore
    .. versionchanged:: 3.10
       Removed the *loop* parameter.
 
+
+Barrier
+=======
+
+   .. versionadded:: 3.11
+
+   A barrier object.  Not thread-safe.
+
+   A barrier is a simple synchronization primitive that allows to block until a
+   certain number of tasks are waiting on it.
+   Tasks can wait on the :meth:`~Barrier.wait` method and would be blocked until
+   the specified number of tasks end up waiting on :meth:`~Barrier.wait`.
+   At that point all of the waiting tasks would unblock simultaneously.
+
+   The barrier can be reused any number of times.
+
+   .. _asyncio_example_barrier:
+
+   Example::
+
+      async def example_barrier():
+         # action when barrier will unblock
+         async def done():
+            print("calling action 'done'")
+
+         # barrier with 3 parties
+         b = asyncio.Barrier(3, done)
+
+         # create 2 new waiting tasks
+         asyncio.create_task(b.wait())
+         asyncio.create_task(b.wait())
+
+         await asyncio.sleep(0)
+         print(b)
+
+         # Now, current task waits
+         async with b as pos:
+            print("barrier passed")
+
+         await asyncio.sleep(0)
+         print(b)
+
+   Result of this example is:
+
+      >>> asyncio.run(example_barrier())
+      <asyncio.locks.Barrier object at 0x103e58bf0 [unlocked, wait:2/3, state:0]>
+      calling action 'done'
+      barrier passed
+      <asyncio.locks.Barrier object at 0x103e58bf0 [unlocked, state:0]>
+
+   As shown in this example, the :meth:`~Barrier.wait` method: can also be call with
+   :keyword:`async with` statement.
+
+   The barrier can be reused any number of times for the same number of tasks.
```

Comment:
Drop this sentence, we already have it few lines earlier.

Reply:
Done